### ISSUE TITLE
Toaplan 1 games

### ISSUE THREAD

**Issue by arcadez** [libretro/mame2003-plus-libretro#371]
Some great games on this system unfortunately on the xbox at least a few of em had
some framerate issues severe stutter and slowdowns in ole MAME, i put it down
to the vblank and fps settings in the drivers as the same games were fine when we 
used FBA.

But try as i might i could never sort this totally until today when i found some
new video code in ShmupMAME which gets all the games upto full speed and keeps
them there

Now the worst offenders are Fire Shark / Same Same Same, Truxton / Tatsujin, Vimana
and to a lesser extent Rally Bike, they might be fine in this core but please give em a whirl 
and lemmie know if we need the faster video code here as well, as i dont wanna go ahead
and port it if it's not required.

Regards

**Comment by grant2258** [comment]
will compile it and get back to you on the framerates mr arcadez

**Comment by grant2258** [comment]
getting between 55 and 52 fps on fireshark, same same seems a little more stable between 56 and 57, you can tell its slowing down the way the sound is going on both and im using the fastest pi 3b+ , rally bike seems ok for me on the pi 3b+anyway. I would say the fireshark/same same would really benefit mr arcadez

**Comment by arcadez** [comment]
Nothing like the drops we got on the Xbox, Fire Shark and Truxton were the worst
Vimana depending on how many enemies were on the screen Rally Bike was ok
although the video was a bit stuttery.

Right thanks for testing i'll go ahead and drop in the special video code from Shmupmame
shortly

**Comment by grant2258** [comment]
will test it when its done my good man.

**Comment by arcadez** [comment]
Well i just reworked and backported the code really, the real graft was done by another dev
@grant2258 BTW did you or anyone get a chance to test DoDonPachi Dai-Ou-Jou, ESPGaluda
and Ketsui..?? i just need to know if they work or not

**Comment by Wilstorm** [comment]
@arcadez - Thanks for the the new games! Do you have the ROM names that need tested?

**Comment by grant2258** [comment]
teste all are working i havent tested  Dai-Ou-Jou ill need to update my game list but all of the above are working some really nice shooters there this and the ssv additions will be keeping me busy for a while. I particularly keen on dondonepatchi and dynagears as well as ultra weapomn x i might be awol for a while !

**Comment by grant2258** [comment]
just looked up the driver  Dai-Ou-Jou is DoDonPachi  its all good mr arcades well is for me on the pi

https://github.com/libretro/mame2003-plus-libretro/blob/3adbd2e0cde31442d7f559ca876d352fa59d2da6/src/drivers/pgm.c#L2522

**Comment by Wilstorm** [comment]
@grant2258 - Ok, I see them, thanks good sir! ;)

**Comment by grant2258** [comment]
Oriental Legend Special - Xi You Shi E Zhuan Super (ver. 101, Korean Board) 
freezes when you put a credit in but it is marked as not working just updating info here

Knights of Valour - Sangoku Senki (ver. 115)
sounds like a squeeky toy when you hit an enemy again this is marked as imperfect sound so jusy info again

Photo Y2K
missing gfx again marke as such in driver so just info

Puzzle Star 
is marked as imperfect sound and graphics the game looks and sounds fine to me maybe im missing something

Knights of Valour Superheroes - Sangoku Senki Superheroes (ver. 322)
mess up gfx and game freezes when you try start it but is marked as not working so just for info again

The Killing Blade (ver. 109, Chinese Board)
have some missing roms will need to fix this

**Comment by grant2258** [comment]
cheers for the rom there is a small problem with the killing blade when you start its fine but when it goes to the Buddha background the sound goes fizzy for lack of better works and works properly again when that background switches botth on windows and the pi. Ill verify the rom as well to make sure its not that and keep you updated on that

**Comment by arcadez** [comment]
Yeah the sound is bad with the original PGM games the Cave ones dont seem as bad im looking
at improving it just now if possible it's just a pity we cant have the good version from later
MAME

**Comment by grant2258** [comment]
its just a small niggle for a short time people can live with that so many other ones work well now I dont think this is a really big one

**Comment by Wilstorm** [comment]
I did test this last night with the same results. I had to wait until I could get to my home PC to rebuild my set. It seems to start when the 2nd round or when the bar turns red and they fight more aggressively. I tried button mashing and that was a no go but for some reason if you wait patiently they are easy to leg sweep until the last bit of red then they are a pain in the ass to take down. =/ Also I couldn't quite tell but during the Buddha background there were black vertical bars. I wasn't sure if that was just part of the flashing-ish background or a niggle.
